pkg/appmanager/appManager.go

```diff
@@ -1785,52 +1787,64 @@ func (appMgr *Manager) deleteUnusedRoutes(namespace, svc string) {
-					poolName := fmt.Sprintf("/%s/%s", cfg.Virtual.Partition, pool.Name)
-					// Delete rule
-					for _, pol := range cfg.Policies {
-						if len(pol.Rules) == 1 {
-							routeName = strings.Split(pol.Rules[0].Name, "_")[3]
-							nr := nameRef{
-								Name:      pol.Name,
-								Partition: pol.Partition,
+				// Make sure we aren't processing empty pool
+				if pool.Name != "" {
+					sKey := serviceKey{
+						ServiceName: pool.ServiceName,
+						ServicePort: pool.ServicePort,
+						Namespace:   namespace,
+					}
+					poolNS := strings.Split(pool.Name, "_")[1]
+					_, ok := appMgr.resources.Get(sKey, cfg.Virtual.VirtualServerName)
+					if pool.ServiceName == svc && poolNS == namespace && !ok {
+						poolName := fmt.Sprintf("/%s/%s", cfg.Virtual.Partition, pool.Name)
+						// Delete rule
+						for _, pol := range cfg.Policies {
+							if len(pol.Rules) == 1 {
+								routeName = strings.Split(pol.Rules[0].Name, "_")[3]
+								nr := nameRef{
+									Name:      pol.Name,
+									Partition: pol.Partition,
+								}
+								cfg.RemovePolicy(nr)
+								continue
 							}
-							cfg.RemovePolicy(nr)
-							continue
-						}
-						for i, rule := range pol.Rules {
-							if len(rule.Actions) > 0 && rule.Actions[0].Pool == poolName {
-								routeName = strings.Split(rule.Name, "_")[3]
-								if i >= len(pol.Rules)-1 {
-									pol.Rules = pol.Rules[:len(pol.Rules)-1]
-								} else {
-									copy(pol.Rules[i:], pol.Rules[i+1:])
-									pol.Rules[len(pol.Rules)-1] = &Rule{}
-									pol.Rules = pol.Rules[:len(pol.Rules)-1]
+							for i, rule := range pol.Rules {
+								if len(rule.Actions) > 0 && rule.Actions[0].Pool == poolName {
+									routeName = strings.Split(rule.Name, "_")[3]
+									if i >= len(pol.Rules)-1 {
+										pol.Rules = pol.Rules[:len(pol.Rules)-1]
+									} else {
+										copy(pol.Rules[i:], pol.Rules[i+1:])
+										pol.Rules[len(pol.Rules)-1] = &Rule{}
+										pol.Rules = pol.Rules[:len(pol.Rules)-1]
+									}
+									cfg.SetPolicy(pol)
 								}
-								cfg.SetPolicy(pol)
 							}
 						}
-					}
-					// Delete pool
-					if i >= len(cfg.Pools)-1 {
-						cfg.Pools = cfg.Pools[:len(cfg.Pools)-1]
-					} else {
-						copy(cfg.Pools[i:], cfg.Pools[i+1:])
-						cfg.Pools[len(cfg.Pools)-1] = Pool{}
-						cfg.Pools = cfg.Pools[:len(cfg.Pools)-1]
-					}
-					// Delete profile
-					if routeName != "" {
-						profileName := fmt.Sprintf("%s/%s-https-cert",
-							cfg.Virtual.Partition, routeName)
-						cfg.Virtual.RemoveFrontendSslProfileName(profileName)
+						// Delete pool
+						if i >= len(cfg.Pools)-1 {
+							cfg.Pools = cfg.Pools[:len(cfg.Pools)-1]
+						} else {
+							copy(cfg.Pools[i:], cfg.Pools[i+1:])
+							cfg.Pools[len(cfg.Pools)-1] = Pool{}
+							cfg.Pools = cfg.Pools[:len(cfg.Pools)-1]
+						}
+						// Delete profile
+						if routeName != "" {
+							client := fmt.Sprintf("%s/openshift_route_%s_%s-client-ssl",
```

Comment:
We should probably have functions that correctly format this name as well as the one for server profiles. I'm concerned that if we ever need to change the name (again) that we will miss some and have hard to find bugs.